api/pkg/apis/v1alpha1/providers/target/helm/helm.go

```diff
@@ -680,26 +689,44 @@ func pullOCIChart(ctx context.Context, repo, version string) (*registry.PullResu
 	return pullRes, nil
 }
 
-func configureInstallClient(name string, componentProps *HelmChartProperty, deployment *model.DeploymentSpec, config *action.Configuration, postRenderer postrender.PostRenderer) *action.Install {
+func configureInstallClient(name string, componentProps *HelmChartProperty, deployment *model.DeploymentSpec, config *action.Configuration, postRenderer postrender.PostRenderer) (*action.Install, error) {
 	installClient := action.NewInstall(config)
 	installClient.ReleaseName = name
 	if deployment.Instance.Spec.Scope == "" {
 		installClient.Namespace = constants.DefaultScope
 	} else {
 		installClient.Namespace = deployment.Instance.Spec.Scope
 	}
+
 	installClient.Wait = componentProps.Wait
+	if componentProps.Timeout != "" {
+		duration, err := time.ParseDuration(componentProps.Timeout)
```

Comment:
It is better we ensure the duration is not negative here. In my limited search time.ParseDuration can support parse (-1m). This should not make sense for helm tool.

Reply:
Helm will throw error :E0828 10:10:01.518232   32780 until.go:159] Timeout for context shall not be negative!

Reply:
I will fix it in next version